charset_reader.go

```diff
@@ -0,0 +1,38 @@
+package excelize
+
+//import "gopkg.in/webnice/debug.v1"
+//import "gopkg.in/webnice/log.v2"
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	"golang.org/x/text/encoding"
+	"golang.org/x/text/encoding/charmap"
+)
+
+// CharsetReader Decoder from all codepages to UTF-8
+func CharsetReader(charset string, input io.Reader) (rdr io.Reader, err error) {
```

Comment:
Please add the tests to cover your changes.

Reply:
ok